{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "from collections import Counter\n",
    "\n",
    "pd.set_option('display.max_rows', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.sorted.length_100.bam',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.artifacts_mapping_stats.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.main.json',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.kmerspecies_61mer_analysis.gene.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.reference_mapping_stats.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.kmerspecies_61mer_analysis.json',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.main.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.allele_mapping_data.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.kmerspecies_61mer_analysis.allele.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.main.kmerspecies_61mer_analysis.json',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.main.kmerspecies_61mer_analysis_rgi_summary.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.sorted.length_100.bam.bai',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.allele_mapping_data.json',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.gene_mapping_data.txt',\n",
       " '../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.overall_mapping_stats.txt']"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# The Gambia study (amr-experiment-1)\n",
    "#prefix = 'output_106_L001_98495_reads_nh' # E. coli\n",
    "#prefix = 'output_133_L001_98477_reads_nh' # E. coli ** \n",
    "#prefix = 'output_107_L001_98461_reads_nh' # E. coli\n",
    "#prefix = 'output_134_L001_98468_reads_nh' # K. pneumoniae **\n",
    "#prefix = 'output_132_L001_98467_reads_nh' # A. baumannii\n",
    "#prefix = 'output_135_L001_98491_reads_nh' # A. baumannii **\n",
    "#prefix = 'output_MI02375-4_S7_L001_74975_reads_nh' # mNGS, patient had isolates from: A. baumannii (x1)\n",
    "#prefix = 'output_MI01672-4_S1_L001_74971_reads_nh' # mNGS, patient had isolates from: E. coli (x2)\n",
    "#prefix = 'output_MI02340-3_S8_L001_74974_reads_nh' # mNGS **, patient had isolates from: E. coli, K. pneumo, A. baumannii\n",
    "\n",
    "#mBAL study\n",
    "#prefix = 'output_mBAL-232-DNA-TA_20357_reads_nh'\n",
    "#prefix = 'output_mBAL-232-RNA-TA_20358_reads_nh'\n",
    "#prefix = 'output_mBAL-239-DNA_20584_reads_nh'\n",
    "#prefix = 'output_mBAL-239-RNA-B5B_20585_reads_nh'\n",
    "#prefix = 'output_mBAL-252-DNA-TA_20371_reads_nh'\n",
    "#prefix = 'output_mBAL-252-RNA-TA_20372_reads_nh'\n",
    "#prefix = 'output_mBAL-257-DNA-TA_20379_reads_nh' # could not merge\n",
    "#prefix = 'output_mBAL-257-RNA-TA_20380_reads_nh'  # could not merge\n",
    "#prefix = 'output_mBAL-277-RNA-B5B_20310_reads_nh'\n",
    "#prefix = 'output_mBAL-297-DNA-TA_20402_reads_nh'\n",
    "#prefix = 'output_mBAL-297-RNA-TA_20403_reads_nh'\n",
    "#prefix = 'output_mBAL-H2O-DNA_libcontrol_OPS030_A5_20311_reads_nh'\n",
    "\n",
    "# GCE cambodia\n",
    "#prefix = 'output_RR176_459_ISOa_S170'\n",
    "\n",
    "#prefix = 'output_RR178_461_ISOa_FLASH_nh'\n",
    "#prefix = 'output_RR180_463_ISOa_FLASH_nh'\n",
    "#prefix = 'output_RR176_459_ISOa_FLASH_nh'\n",
    "#prefix = 'output_RR178_461_ISOa_S111_nh'\n",
    "#prefix = 'output_RR180_463_ISOa_S135_nh'\n",
    "#prefix = 'output_RR176_459_ISOa_S170_nh'\n",
    "\n",
    "# mecA study\n",
    "#prefix = 'output_RR108e_00010_B2_S10_61787_reads_nh'\n",
    "#prefix = 'output_RR108e_00017_A3_S17_61794_reads_nh'\n",
    "#prefix = 'output_RR108e_00071_S1_207817_reads_nh'\n",
    "#prefix = 'output_RR108e_00072_S2_207818_reads_nh'\n",
    "prefix = 'output_RR108e_00073_S3_207819_reads_nh'\n",
    "\n",
    "\n",
    "output_dir = f\"../{prefix}/\"\n",
    "\n",
    "glob.glob(output_dir + '*')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load data! (for main-amr, main-species, kma-amr, kma-species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ORF_ID\n",
      "Contig\n",
      "Start\n",
      "Stop\n",
      "Orientation\n",
      "Cut_Off\n",
      "Pass_Bitscore\n",
      "Best_Hit_Bitscore\n",
      "Best_Hit_ARO\n",
      "Best_Identities\n",
      "ARO\n",
      "Model_type\n",
      "SNPs_in_Best_Hit_ARO\n",
      "Other_SNPs\n",
      "Drug Class\n",
      "Resistance Mechanism\n",
      "AMR Gene Family\n",
      "Predicted_DNA\n",
      "Predicted_Protein\n",
      "CARD_Protein_Sequence\n",
      "Percentage Length of Reference Sequence\n",
      "ID\n",
      "Model_ID\n",
      "Nudged\n",
      "Note\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Cut_Off</th>\n",
       "      <th>Best_Hit_ARO</th>\n",
       "      <th>AMR Gene Family</th>\n",
       "      <th>Model_type</th>\n",
       "      <th>Best_Identities</th>\n",
       "      <th>Percentage Length of Reference Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9 # 11869 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9</td>\n",
       "      <td>Strict</td>\n",
       "      <td>arlr</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>99.54</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8 # 10517 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>arls</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78 # 69097 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepa</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79 # 70559 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepr</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9 # 7790 # ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mgra</td>\n",
       "      <td>ATP-binding cassette (ABC) antibiotic efflux p...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>Strict</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>blaZ beta-lactamase</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>93.95</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>antibiotic-resistant murA transferase</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>99.29</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>99.74</td>\n",
       "      <td>100.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>Strict</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>99.78</td>\n",
       "      <td>100.44</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              ORF_ID  \\\n",
       "8  NODE_47_length_13669_cov_51.565374_9 # 11869 #...   \n",
       "7  NODE_47_length_13669_cov_51.565374_8 # 10517 #...   \n",
       "0  NODE_4_length_124562_cov_63.896729_78 # 69097 ...   \n",
       "1  NODE_4_length_124562_cov_63.896729_79 # 70559 ...   \n",
       "4  NODE_41_length_18243_cov_57.681328_9 # 7790 # ...   \n",
       "6  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "3  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "9  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "5  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "2  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                                   Contig  Cut_Off  \\\n",
       "8    NODE_47_length_13669_cov_51.565374_9   Strict   \n",
       "7    NODE_47_length_13669_cov_51.565374_8  Perfect   \n",
       "0   NODE_4_length_124562_cov_63.896729_78  Perfect   \n",
       "1   NODE_4_length_124562_cov_63.896729_79  Perfect   \n",
       "4    NODE_41_length_18243_cov_57.681328_9  Perfect   \n",
       "6   NODE_46_length_14877_cov_127.795875_5   Strict   \n",
       "3    NODE_37_length_19743_cov_60.568519_2  Perfect   \n",
       "9  NODE_13_length_74983_cov_65.352979_38    Strict   \n",
       "5   NODE_41_length_18243_cov_57.681328_17   Strict   \n",
       "2    NODE_32_length_28365_cov_66.015789_2   Strict   \n",
       "\n",
       "                                        Best_Hit_ARO  \\\n",
       "8                                               arlr   \n",
       "7                                               arls   \n",
       "0                                               mepa   \n",
       "1                                               mepr   \n",
       "4                                               mgra   \n",
       "6                          pc1 beta-lactamase (blaz)   \n",
       "3                         staphylococcus aureus lmrs   \n",
       "9  staphylococcus aureus mura with mutation confe...   \n",
       "5                         staphylococcus aureus nora   \n",
       "2                                            tet(38)   \n",
       "\n",
       "                                     AMR Gene Family             Model_type  \\\n",
       "8  major facilitator superfamily (MFS) antibiotic...  protein homolog model   \n",
       "7  major facilitator superfamily (MFS) antibiotic...  protein homolog model   \n",
       "0  multidrug and toxic compound extrusion (MATE) ...  protein homolog model   \n",
       "1  multidrug and toxic compound extrusion (MATE) ...  protein homolog model   \n",
       "4  ATP-binding cassette (ABC) antibiotic efflux p...  protein homolog model   \n",
       "6                                blaZ beta-lactamase  protein homolog model   \n",
       "3  major facilitator superfamily (MFS) antibiotic...  protein homolog model   \n",
       "9              antibiotic-resistant murA transferase  protein variant model   \n",
       "5  major facilitator superfamily (MFS) antibiotic...  protein homolog model   \n",
       "2  major facilitator superfamily (MFS) antibiotic...  protein homolog model   \n",
       "\n",
       "   Best_Identities  Percentage Length of Reference Sequence  \n",
       "8            99.54                                   100.00  \n",
       "7           100.00                                   100.00  \n",
       "0           100.00                                   100.00  \n",
       "1           100.00                                   100.00  \n",
       "4           100.00                                   100.00  \n",
       "6            93.95                                   100.00  \n",
       "3           100.00                                   100.21  \n",
       "9            99.29                                   100.00  \n",
       "5            99.74                                   100.00  \n",
       "2            99.78                                   100.44  "
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# format the main-amr findings\n",
    "#\n",
    "\n",
    "# read in the RGI main AMR results\n",
    "main_output = pd.read_csv(f\"{output_dir}{prefix}.rgi.main.txt\", sep='\\t')\n",
    "print('\\n'.join(main_output.columns)) #print names of columns\n",
    "\n",
    "# convert Best_Hit_ARO to all lowercase for matching purposes\n",
    "main_output['Best_Hit_ARO'] = [i.lower() for i in main_output['Best_Hit_ARO']] \n",
    "main_output.sort_values(by='Best_Hit_ARO', inplace=True)\n",
    "main_output.reset_index()\n",
    "\n",
    "# show subset of dataframe for manual checking visualization purposes\n",
    "main_output[['ORF_ID', 'Contig', 'Cut_Off', 'Best_Hit_ARO', 'AMR Gene Family', 'Model_type', 'Best_Identities', 'Percentage Length of Reference Sequence']]#.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Cut_Off</th>\n",
       "      <th>Best_Hit_ARO</th>\n",
       "      <th>CARD*kmer Prediction</th>\n",
       "      <th>Taxonomic kmers</th>\n",
       "      <th>Genomic kmers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9 # 11869 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9</td>\n",
       "      <td>Strict</td>\n",
       "      <td>arlr</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 73;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 539;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8 # 10517 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>arls</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 574;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1296;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78 # 69097 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepa</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 455;</td>\n",
       "      <td>chr + plasmid: 1296; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79 # 70559 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepr</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 318;</td>\n",
       "      <td>chr + plasmid: 360; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9 # 7790 # ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mgra</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 196;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 384;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>Strict</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 170;</td>\n",
       "      <td>chr + plasmid: 770; plasmid: 16; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 914;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1386;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 1206;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1206;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 656;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1107;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>Strict</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 1109;</td>\n",
       "      <td>chr + plasmid: 1299; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              ORF_ID  \\\n",
       "8  NODE_47_length_13669_cov_51.565374_9 # 11869 #...   \n",
       "7  NODE_47_length_13669_cov_51.565374_8 # 10517 #...   \n",
       "0  NODE_4_length_124562_cov_63.896729_78 # 69097 ...   \n",
       "1  NODE_4_length_124562_cov_63.896729_79 # 70559 ...   \n",
       "4  NODE_41_length_18243_cov_57.681328_9 # 7790 # ...   \n",
       "6  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "3  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "9  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "5  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "2  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                                  Contig  Cut_Off  \\\n",
       "8   NODE_47_length_13669_cov_51.565374_9   Strict   \n",
       "7   NODE_47_length_13669_cov_51.565374_8  Perfect   \n",
       "0  NODE_4_length_124562_cov_63.896729_78  Perfect   \n",
       "1  NODE_4_length_124562_cov_63.896729_79  Perfect   \n",
       "4   NODE_41_length_18243_cov_57.681328_9  Perfect   \n",
       "6  NODE_46_length_14877_cov_127.795875_5   Strict   \n",
       "3   NODE_37_length_19743_cov_60.568519_2  Perfect   \n",
       "9  NODE_13_length_74983_cov_65.352979_38   Strict   \n",
       "5  NODE_41_length_18243_cov_57.681328_17   Strict   \n",
       "2   NODE_32_length_28365_cov_66.015789_2   Strict   \n",
       "\n",
       "                                        Best_Hit_ARO  \\\n",
       "8                                               arlr   \n",
       "7                                               arls   \n",
       "0                                               mepa   \n",
       "1                                               mepr   \n",
       "4                                               mgra   \n",
       "6                          pc1 beta-lactamase (blaz)   \n",
       "3                         staphylococcus aureus lmrs   \n",
       "9  staphylococcus aureus mura with mutation confe...   \n",
       "5                         staphylococcus aureus nora   \n",
       "2                                            tet(38)   \n",
       "\n",
       "                            CARD*kmer Prediction  \\\n",
       "8             Staphylococcus aureus (chromosome)   \n",
       "7             Staphylococcus aureus (chromosome)   \n",
       "0  Staphylococcus aureus (chromosome or plasmid)   \n",
       "1  Staphylococcus aureus (chromosome or plasmid)   \n",
       "4             Staphylococcus aureus (chromosome)   \n",
       "6  Staphylococcus aureus (chromosome or plasmid)   \n",
       "3             Staphylococcus aureus (chromosome)   \n",
       "9             Staphylococcus aureus (chromosome)   \n",
       "5             Staphylococcus aureus (chromosome)   \n",
       "2  Staphylococcus aureus (chromosome or plasmid)   \n",
       "\n",
       "                 Taxonomic kmers                              Genomic kmers  \n",
       "8    Staphylococcus aureus: 73;    chr + plasmid: 0; plasmid: 0; chr: 539;   \n",
       "7   Staphylococcus aureus: 574;   chr + plasmid: 0; plasmid: 0; chr: 1296;   \n",
       "0   Staphylococcus aureus: 455;   chr + plasmid: 1296; plasmid: 0; chr: 0;   \n",
       "1   Staphylococcus aureus: 318;    chr + plasmid: 360; plasmid: 0; chr: 0;   \n",
       "4   Staphylococcus aureus: 196;    chr + plasmid: 0; plasmid: 0; chr: 384;   \n",
       "6   Staphylococcus aureus: 170;   chr + plasmid: 770; plasmid: 16; chr: 0;   \n",
       "3   Staphylococcus aureus: 914;   chr + plasmid: 0; plasmid: 0; chr: 1386;   \n",
       "9  Staphylococcus aureus: 1206;   chr + plasmid: 0; plasmid: 0; chr: 1206;   \n",
       "5   Staphylococcus aureus: 656;   chr + plasmid: 0; plasmid: 0; chr: 1107;   \n",
       "2  Staphylococcus aureus: 1109;   chr + plasmid: 1299; plasmid: 0; chr: 0;   "
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# format the main-species findings\n",
    "#\n",
    "\n",
    "# read in the RGI main SPECIES-calling results\n",
    "main_species_output = pd.read_csv(f\"{output_dir}{prefix}.rgi.main.kmerspecies_61mer_analysis_rgi_summary.txt\", sep='\\t')\n",
    "\n",
    "# convert Best_Hit_ARO to lowercase for matching purposes\n",
    "main_species_output['Best_Hit_ARO'] = [i.lower() for i in main_species_output['Best_Hit_ARO']] \n",
    "main_species_output.sort_values(by = 'Best_Hit_ARO', inplace=True)\n",
    "main_species_output.reset_index()\n",
    "main_species_output.head()   # show dataframe for manual checking visualization purposes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reference Sequence\n",
      "ARO Term\n",
      "ARO Accession\n",
      "Reference Model Type\n",
      "Reference DB\n",
      "Reference Allele Source\n",
      "Resistomes & Variants: Observed in Genome(s)\n",
      "Resistomes & Variants: Observed in Plasmid(s)\n",
      "Resistomes & Variants: Observed Pathogen(s)\n",
      "Completely Mapped Reads\n",
      "Mapped Reads with Flanking Sequence\n",
      "All Mapped Reads\n",
      "Percent Coverage\n",
      "Length Coverage (bp)\n",
      "Average MAPQ (Completely Mapped Reads)\n",
      "Mate Pair Linkage\n",
      "Reference Length\n",
      "AMR Gene Family\n",
      "Drug Class\n",
      "Resistance Mechanism\n",
      "Depth\n",
      "SNPs\n",
      "Consensus Sequence DNA\n",
      "Consensus Sequence Protein\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARO Term</th>\n",
       "      <th>AMR Gene Family</th>\n",
       "      <th>ARO Accession</th>\n",
       "      <th>Reference Model Type</th>\n",
       "      <th>Resistomes &amp; Variants: Observed Pathogen(s)</th>\n",
       "      <th>Percent Coverage</th>\n",
       "      <th>Depth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>arlr</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>3000838</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>75.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arls</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>3000839</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>73.89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>borreliella burgdorferi 16s rrna mutation conf...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003504</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Borreliella burgdorferi</td>\n",
       "      <td>50.64</td>\n",
       "      <td>29.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>cfrc</td>\n",
       "      <td>Cfr 23S ribosomal RNA methyltransferase</td>\n",
       "      <td>3004146</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Clostridium botulinum</td>\n",
       "      <td>1.84</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chlamydophila psittaci 16s rrna mutation confe...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003485</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Chlamydia psittaci</td>\n",
       "      <td>32.58</td>\n",
       "      <td>0.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>cutibacterium acnes 16s rrna mutation conferri...</td>\n",
       "      <td>16S rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003499</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Cutibacterium acnes</td>\n",
       "      <td>44.82</td>\n",
       "      <td>0.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>enterococcus faecium ef-tu mutants conferring ...</td>\n",
       "      <td>elfamycin resistant EF-Tu</td>\n",
       "      <td>3003438</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Enterococcus faecium</td>\n",
       "      <td>72.48</td>\n",
       "      <td>22.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>escherichia coli 16s rrna mutation conferring ...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003223</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>68.74</td>\n",
       "      <td>11.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>helicobacter pylori 16s rrna mutation conferri...</td>\n",
       "      <td>16S rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003510</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Helicobacter pylori</td>\n",
       "      <td>78.48</td>\n",
       "      <td>51.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>mecc-type blaz</td>\n",
       "      <td>blaZ beta-lactamase</td>\n",
       "      <td>3005097</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>113.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mepa</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>3000026</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>113.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mepr</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>3000746</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>117.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mgra</td>\n",
       "      <td>ATP-binding cassette (ABC) antibiotic efflux p...</td>\n",
       "      <td>3000815</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>114.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>mycobacteroides abscessus 16s rrna mutation co...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003240</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Mycobacteroides abscessus</td>\n",
       "      <td>11.70</td>\n",
       "      <td>0.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna mutation ...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003541</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Mycolicibacterium smegmatis</td>\n",
       "      <td>84.10</td>\n",
       "      <td>49.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>neisseria gonorrhoeae 16s rrna mutation confer...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003495</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Neisseria gonorrhoeae</td>\n",
       "      <td>64.35</td>\n",
       "      <td>9.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>pasteurella multocida 16s rrna mutation confer...</td>\n",
       "      <td>16s rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3003493</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Pasteurella multocida</td>\n",
       "      <td>33.85</td>\n",
       "      <td>1.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>blaZ beta-lactamase</td>\n",
       "      <td>3000621</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>286.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>staphylococcus aureus 23s rrna with mutation c...</td>\n",
       "      <td>23S rRNA with mutation conferring resistance t...</td>\n",
       "      <td>3004058</td>\n",
       "      <td>rRNA gene variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>634.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>staphylococcus aureus cls conferring resistanc...</td>\n",
       "      <td>daptomycin resistant cls</td>\n",
       "      <td>3003074</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>94.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>staphylococcus aureus fosb</td>\n",
       "      <td>fosfomycin thiol transferase</td>\n",
       "      <td>3004661</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>101.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>staphylococcus aureus glpt with mutation confe...</td>\n",
       "      <td>antibiotic-resistant GlpT</td>\n",
       "      <td>3003901</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>124.77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>staphylococcus aureus gyra conferring resistan...</td>\n",
       "      <td>fluoroquinolone resistant gyrA</td>\n",
       "      <td>3003296</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>117.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>staphylococcus aureus gyrb conferring resistan...</td>\n",
       "      <td>aminocoumarin resistant gyrB</td>\n",
       "      <td>3003301</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>121.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>staphylococcus aureus iles with mutation confe...</td>\n",
       "      <td>antibiotic-resistant isoleucyl-tRNA synthetase...</td>\n",
       "      <td>3003729</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>99.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>3004572</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>100.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>staphylococcus aureus mena with mutation confe...</td>\n",
       "      <td>lysocin resistant menA</td>\n",
       "      <td>3003917</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>107.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>staphylococcus aureus mprf mutations conferrin...</td>\n",
       "      <td>daptomycin resistant mprF</td>\n",
       "      <td>3003319</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>80.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>antibiotic-resistant murA transferase</td>\n",
       "      <td>3003776</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>93.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>3004667</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>98.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "      <td>fluoroquinolone resistant parC</td>\n",
       "      <td>3003312</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>82.62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>staphylococcus aureus pare conferring resistan...</td>\n",
       "      <td>aminocoumarin resistant parE</td>\n",
       "      <td>3003314</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>99.95</td>\n",
       "      <td>88.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>staphylococcus aureus pare conferring resistan...</td>\n",
       "      <td>fluoroquinolone resistant parE</td>\n",
       "      <td>3003315</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>37.94</td>\n",
       "      <td>7.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>staphylococcus aureus pgsa mutations conferrin...</td>\n",
       "      <td>daptomycin resistant pgsA</td>\n",
       "      <td>3003323</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>77.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>staphylococcus aureus rpob mutants conferring ...</td>\n",
       "      <td>rifamycin-resistant beta-subunit of RNA polyme...</td>\n",
       "      <td>3003285</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>109.61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>staphylococcus aureus rpoc conferring resistan...</td>\n",
       "      <td>daptomycin-resistant beta prime subunit of RNA...</td>\n",
       "      <td>3003291</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>113.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>staphylococcus aureus uhpt with mutation confe...</td>\n",
       "      <td>antibiotic-resistant UhpT</td>\n",
       "      <td>3003902</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>117.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "      <td>daptomycin resistant walK</td>\n",
       "      <td>3003794</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>118.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>staphylococcus mutant fusa gene conferring res...</td>\n",
       "      <td>antibiotic resistant fusA</td>\n",
       "      <td>3003735</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>109.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>staphylococcus mutant fuse gene conferring res...</td>\n",
       "      <td>antibiotic resistant fusE</td>\n",
       "      <td>3003737</td>\n",
       "      <td>protein variant model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>111.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>tet(38)</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>3000565</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>100.00</td>\n",
       "      <td>107.01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             ARO Term  \\\n",
       "5                                                arlr   \n",
       "1                                                arls   \n",
       "21  borreliella burgdorferi 16s rrna mutation conf...   \n",
       "38                                               cfrc   \n",
       "20  chlamydophila psittaci 16s rrna mutation confe...   \n",
       "24  cutibacterium acnes 16s rrna mutation conferri...   \n",
       "37  enterococcus faecium ef-tu mutants conferring ...   \n",
       "15  escherichia coli 16s rrna mutation conferring ...   \n",
       "35  helicobacter pylori 16s rrna mutation conferri...   \n",
       "39                                     mecc-type blaz   \n",
       "2                                                mepa   \n",
       "4                                                mepr   \n",
       "0                                                mgra   \n",
       "22  mycobacteroides abscessus 16s rrna mutation co...   \n",
       "23  mycolicibacterium smegmatis 16s rrna mutation ...   \n",
       "16  neisseria gonorrhoeae 16s rrna mutation confer...   \n",
       "19  pasteurella multocida 16s rrna mutation confer...   \n",
       "3                           pc1 beta-lactamase (blaz)   \n",
       "36  staphylococcus aureus 23s rrna with mutation c...   \n",
       "13  staphylococcus aureus cls conferring resistanc...   \n",
       "33                         staphylococcus aureus fosb   \n",
       "31  staphylococcus aureus glpt with mutation confe...   \n",
       "12  staphylococcus aureus gyra conferring resistan...   \n",
       "14  staphylococcus aureus gyrb conferring resistan...   \n",
       "25  staphylococcus aureus iles with mutation confe...   \n",
       "40                         staphylococcus aureus lmrs   \n",
       "30  staphylococcus aureus mena with mutation confe...   \n",
       "11  staphylococcus aureus mprf mutations conferrin...   \n",
       "28  staphylococcus aureus mura with mutation confe...   \n",
       "34                         staphylococcus aureus nora   \n",
       "17  staphylococcus aureus parc conferring resistan...   \n",
       "18  staphylococcus aureus pare conferring resistan...   \n",
       "8   staphylococcus aureus pare conferring resistan...   \n",
       "9   staphylococcus aureus pgsa mutations conferrin...   \n",
       "7   staphylococcus aureus rpob mutants conferring ...   \n",
       "10  staphylococcus aureus rpoc conferring resistan...   \n",
       "32  staphylococcus aureus uhpt with mutation confe...   \n",
       "29  staphylococcus aureus walk with mutation confe...   \n",
       "26  staphylococcus mutant fusa gene conferring res...   \n",
       "27  staphylococcus mutant fuse gene conferring res...   \n",
       "6                                             tet(38)   \n",
       "\n",
       "                                      AMR Gene Family  ARO Accession  \\\n",
       "5   major facilitator superfamily (MFS) antibiotic...        3000838   \n",
       "1   major facilitator superfamily (MFS) antibiotic...        3000839   \n",
       "21  16s rRNA with mutation conferring resistance t...        3003504   \n",
       "38            Cfr 23S ribosomal RNA methyltransferase        3004146   \n",
       "20  16s rRNA with mutation conferring resistance t...        3003485   \n",
       "24  16S rRNA with mutation conferring resistance t...        3003499   \n",
       "37                          elfamycin resistant EF-Tu        3003438   \n",
       "15  16s rRNA with mutation conferring resistance t...        3003223   \n",
       "35  16S rRNA with mutation conferring resistance t...        3003510   \n",
       "39                                blaZ beta-lactamase        3005097   \n",
       "2   multidrug and toxic compound extrusion (MATE) ...        3000026   \n",
       "4   multidrug and toxic compound extrusion (MATE) ...        3000746   \n",
       "0   ATP-binding cassette (ABC) antibiotic efflux p...        3000815   \n",
       "22  16s rRNA with mutation conferring resistance t...        3003240   \n",
       "23  16s rRNA with mutation conferring resistance t...        3003541   \n",
       "16  16s rRNA with mutation conferring resistance t...        3003495   \n",
       "19  16s rRNA with mutation conferring resistance t...        3003493   \n",
       "3                                 blaZ beta-lactamase        3000621   \n",
       "36  23S rRNA with mutation conferring resistance t...        3004058   \n",
       "13                           daptomycin resistant cls        3003074   \n",
       "33                       fosfomycin thiol transferase        3004661   \n",
       "31                          antibiotic-resistant GlpT        3003901   \n",
       "12                     fluoroquinolone resistant gyrA        3003296   \n",
       "14                       aminocoumarin resistant gyrB        3003301   \n",
       "25  antibiotic-resistant isoleucyl-tRNA synthetase...        3003729   \n",
       "40  major facilitator superfamily (MFS) antibiotic...        3004572   \n",
       "30                             lysocin resistant menA        3003917   \n",
       "11                          daptomycin resistant mprF        3003319   \n",
       "28              antibiotic-resistant murA transferase        3003776   \n",
       "34  major facilitator superfamily (MFS) antibiotic...        3004667   \n",
       "17                     fluoroquinolone resistant parC        3003312   \n",
       "18                       aminocoumarin resistant parE        3003314   \n",
       "8                      fluoroquinolone resistant parE        3003315   \n",
       "9                           daptomycin resistant pgsA        3003323   \n",
       "7   rifamycin-resistant beta-subunit of RNA polyme...        3003285   \n",
       "10  daptomycin-resistant beta prime subunit of RNA...        3003291   \n",
       "32                          antibiotic-resistant UhpT        3003902   \n",
       "29                          daptomycin resistant walK        3003794   \n",
       "26                          antibiotic resistant fusA        3003735   \n",
       "27                          antibiotic resistant fusE        3003737   \n",
       "6   major facilitator superfamily (MFS) antibiotic...        3000565   \n",
       "\n",
       "       Reference Model Type Resistomes & Variants: Observed Pathogen(s)  \\\n",
       "5     protein homolog model                       Staphylococcus aureus   \n",
       "1     protein homolog model                       Staphylococcus aureus   \n",
       "21  rRNA gene variant model                     Borreliella burgdorferi   \n",
       "38    protein homolog model                       Clostridium botulinum   \n",
       "20  rRNA gene variant model                          Chlamydia psittaci   \n",
       "24  rRNA gene variant model                         Cutibacterium acnes   \n",
       "37    protein variant model                        Enterococcus faecium   \n",
       "15  rRNA gene variant model                            Escherichia coli   \n",
       "35  rRNA gene variant model                         Helicobacter pylori   \n",
       "39    protein homolog model                       Staphylococcus aureus   \n",
       "2     protein homolog model                       Staphylococcus aureus   \n",
       "4     protein homolog model                       Staphylococcus aureus   \n",
       "0     protein homolog model                       Staphylococcus aureus   \n",
       "22  rRNA gene variant model                   Mycobacteroides abscessus   \n",
       "23  rRNA gene variant model                 Mycolicibacterium smegmatis   \n",
       "16  rRNA gene variant model                       Neisseria gonorrhoeae   \n",
       "19  rRNA gene variant model                       Pasteurella multocida   \n",
       "3     protein homolog model                       Staphylococcus aureus   \n",
       "36  rRNA gene variant model                       Staphylococcus aureus   \n",
       "13    protein variant model                       Staphylococcus aureus   \n",
       "33    protein homolog model                       Staphylococcus aureus   \n",
       "31    protein variant model                       Staphylococcus aureus   \n",
       "12    protein variant model                              Staphylococcus   \n",
       "14    protein variant model                       Staphylococcus aureus   \n",
       "25    protein variant model                       Staphylococcus aureus   \n",
       "40    protein homolog model                       Staphylococcus aureus   \n",
       "30    protein variant model                       Staphylococcus aureus   \n",
       "11    protein variant model                       Staphylococcus aureus   \n",
       "28    protein variant model                       Staphylococcus aureus   \n",
       "34    protein homolog model                       Staphylococcus aureus   \n",
       "17    protein variant model                              Staphylococcus   \n",
       "18    protein variant model                       Staphylococcus aureus   \n",
       "8     protein variant model                       Staphylococcus aureus   \n",
       "9     protein variant model                       Staphylococcus aureus   \n",
       "7     protein variant model                       Staphylococcus aureus   \n",
       "10    protein variant model                       Staphylococcus aureus   \n",
       "32    protein variant model                       Staphylococcus aureus   \n",
       "29    protein variant model                       Staphylococcus aureus   \n",
       "26    protein variant model                       Staphylococcus aureus   \n",
       "27    protein variant model                       Staphylococcus aureus   \n",
       "6     protein homolog model                       Staphylococcus aureus   \n",
       "\n",
       "    Percent Coverage   Depth  \n",
       "5             100.00   75.80  \n",
       "1             100.00   73.89  \n",
       "21             50.64   29.57  \n",
       "38              1.84    0.02  \n",
       "20             32.58    0.78  \n",
       "24             44.82    0.20  \n",
       "37             72.48   22.88  \n",
       "15             68.74   11.81  \n",
       "35             78.48   51.23  \n",
       "39            100.00  113.12  \n",
       "2             100.00  113.22  \n",
       "4             100.00  117.15  \n",
       "0             100.00  114.67  \n",
       "22             11.70    0.19  \n",
       "23             84.10   49.74  \n",
       "16             64.35    9.93  \n",
       "19             33.85    1.60  \n",
       "3             100.00  286.91  \n",
       "36            100.00  634.03  \n",
       "13            100.00   94.18  \n",
       "33            100.00  101.92  \n",
       "31            100.00  124.77  \n",
       "12            100.00  117.08  \n",
       "14            100.00  121.06  \n",
       "25            100.00   99.02  \n",
       "40            100.00  100.01  \n",
       "30            100.00  107.66  \n",
       "11            100.00   80.14  \n",
       "28            100.00   93.54  \n",
       "34            100.00   98.34  \n",
       "17            100.00   82.62  \n",
       "18             99.95   88.30  \n",
       "8              37.94    7.85  \n",
       "9             100.00   77.14  \n",
       "7             100.00  109.61  \n",
       "10            100.00  113.59  \n",
       "32            100.00  117.32  \n",
       "29            100.00  118.82  \n",
       "26            100.00  109.49  \n",
       "27            100.00  111.36  \n",
       "6             100.00  107.01  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# format the KMA-amr findings\n",
    "#\n",
    "\n",
    "# read in the results of KMA AMR \n",
    "kma_output = pd.read_csv(f\"{output_dir}{prefix}.rgi.kma.allele_mapping_data.txt\", sep='\\t')\n",
    "\n",
    "# convert 'ARO Term' to lowercase for matching\n",
    "kma_output['ARO Term'] = [i.lower() for i in kma_output['ARO Term']] \n",
    "kma_output.sort_values(by = 'ARO Term', inplace=True)\n",
    "print('\\n'.join(kma_output.columns))  # print names of columns\n",
    "\n",
    "kma_output.reset_index()\n",
    "\n",
    "# display subset of data for manual visual-checking\n",
    "kma_output[['ARO Term', 'AMR Gene Family', 'ARO Accession', 'Reference Model Type', 'Resistomes & Variants: Observed Pathogen(s)', 'Percent Coverage', 'Depth']]#.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../output_RR108e_00073_S3_207819_reads_nh/output_RR108e_00073_S3_207819_reads_nh.rgi.kma.kmerspecies_61mer_analysis.gene.txt\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ARO term</th>\n",
       "      <th>CARD*kmer Prediction</th>\n",
       "      <th>Mapped reads with kmer DB hits</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>arlr</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 130; Staph...</td>\n",
       "      <td>370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arls</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 405; Staph...</td>\n",
       "      <td>735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>cutibacterium acnes 16s rrna mutation conferri...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>helicobacter pylori 16s rrna mutation conferri...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mepa</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid):...</td>\n",
       "      <td>1102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mepr</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid):...</td>\n",
       "      <td>341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mgra</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 258;</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>mycobacteroides abscessus 16s rrna mutation co...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna (rrsb) mu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid):...</td>\n",
       "      <td>1817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>staphylococcus aureus 23s rrna with mutation c...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 6335; Chla...</td>\n",
       "      <td>13167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>staphylococcus aureus cls conferring resistanc...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 990;</td>\n",
       "      <td>990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>staphylococcus aureus fosb</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 310;</td>\n",
       "      <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>staphylococcus aureus fusa with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1621;</td>\n",
       "      <td>1621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>staphylococcus aureus glpt with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 889;</td>\n",
       "      <td>1215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>staphylococcus aureus gyra conferring resistan...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 8; Staphyl...</td>\n",
       "      <td>2234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>staphylococcus aureus gyrb conferring resistan...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>staphylococcus aureus iles with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1948;</td>\n",
       "      <td>1948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 820;</td>\n",
       "      <td>1047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>staphylococcus aureus mprf with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1465;</td>\n",
       "      <td>1465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 876;</td>\n",
       "      <td>876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 648;</td>\n",
       "      <td>825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1284;</td>\n",
       "      <td>1430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>staphylococcus aureus pare conferring resistan...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1242;</td>\n",
       "      <td>1242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>staphylococcus aureus pare conferring resistan...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 104;</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>staphylococcus aureus rpob mutants conferring ...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 2776;</td>\n",
       "      <td>2776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>staphylococcus aureus uhpt with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1062;</td>\n",
       "      <td>1062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome): 1486;</td>\n",
       "      <td>1486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>tet(38)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid):...</td>\n",
       "      <td>1043</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             ARO term  \\\n",
       "5                                                arlr   \n",
       "1                                                arls   \n",
       "17  cutibacterium acnes 16s rrna mutation conferri...   \n",
       "26  helicobacter pylori 16s rrna mutation conferri...   \n",
       "2                                                mepa   \n",
       "4                                                mepr   \n",
       "0                                                mgra   \n",
       "15  mycobacteroides abscessus 16s rrna mutation co...   \n",
       "16  mycolicibacterium smegmatis 16s rrna (rrsb) mu...   \n",
       "3                           pc1 beta-lactamase (blaz)   \n",
       "27  staphylococcus aureus 23s rrna with mutation c...   \n",
       "11  staphylococcus aureus cls conferring resistanc...   \n",
       "24                         staphylococcus aureus fosb   \n",
       "19  staphylococcus aureus fusa with mutation confe...   \n",
       "22  staphylococcus aureus glpt with mutation confe...   \n",
       "10  staphylococcus aureus gyra conferring resistan...   \n",
       "12  staphylococcus aureus gyrb conferring resistan...   \n",
       "18  staphylococcus aureus iles with mutation confe...   \n",
       "28                         staphylococcus aureus lmrs   \n",
       "9   staphylococcus aureus mprf with mutation confe...   \n",
       "20  staphylococcus aureus mura with mutation confe...   \n",
       "25                         staphylococcus aureus nora   \n",
       "13  staphylococcus aureus parc conferring resistan...   \n",
       "14  staphylococcus aureus pare conferring resistan...   \n",
       "8   staphylococcus aureus pare conferring resistan...   \n",
       "7   staphylococcus aureus rpob mutants conferring ...   \n",
       "23  staphylococcus aureus uhpt with mutation confe...   \n",
       "21  staphylococcus aureus walk with mutation confe...   \n",
       "6                                             tet(38)   \n",
       "\n",
       "                                 CARD*kmer Prediction  \\\n",
       "5   Staphylococcus aureus (chromosome): 130; Staph...   \n",
       "1   Staphylococcus aureus (chromosome): 405; Staph...   \n",
       "17                                                NaN   \n",
       "26                                                NaN   \n",
       "2   Staphylococcus aureus (chromosome or plasmid):...   \n",
       "4   Staphylococcus aureus (chromosome or plasmid):...   \n",
       "0           Staphylococcus aureus (chromosome): 258;    \n",
       "15                                                NaN   \n",
       "16                                                NaN   \n",
       "3   Staphylococcus aureus (chromosome or plasmid):...   \n",
       "27  Staphylococcus aureus (chromosome): 6335; Chla...   \n",
       "11          Staphylococcus aureus (chromosome): 990;    \n",
       "24          Staphylococcus aureus (chromosome): 310;    \n",
       "19         Staphylococcus aureus (chromosome): 1621;    \n",
       "22          Staphylococcus aureus (chromosome): 889;    \n",
       "10  Staphylococcus aureus (chromosome): 8; Staphyl...   \n",
       "12                                                NaN   \n",
       "18         Staphylococcus aureus (chromosome): 1948;    \n",
       "28          Staphylococcus aureus (chromosome): 820;    \n",
       "9          Staphylococcus aureus (chromosome): 1465;    \n",
       "20          Staphylococcus aureus (chromosome): 876;    \n",
       "25          Staphylococcus aureus (chromosome): 648;    \n",
       "13         Staphylococcus aureus (chromosome): 1284;    \n",
       "14         Staphylococcus aureus (chromosome): 1242;    \n",
       "8           Staphylococcus aureus (chromosome): 104;    \n",
       "7          Staphylococcus aureus (chromosome): 2776;    \n",
       "23         Staphylococcus aureus (chromosome): 1062;    \n",
       "21         Staphylococcus aureus (chromosome): 1486;    \n",
       "6   Staphylococcus aureus (chromosome or plasmid):...   \n",
       "\n",
       "    Mapped reads with kmer DB hits  \n",
       "5                              370  \n",
       "1                              735  \n",
       "17                               0  \n",
       "26                               0  \n",
       "2                             1102  \n",
       "4                              341  \n",
       "0                              364  \n",
       "15                               0  \n",
       "16                               0  \n",
       "3                             1817  \n",
       "27                           13167  \n",
       "11                             990  \n",
       "24                             311  \n",
       "19                            1621  \n",
       "22                            1215  \n",
       "10                            2234  \n",
       "12                               0  \n",
       "18                            1948  \n",
       "28                            1047  \n",
       "9                             1465  \n",
       "20                             876  \n",
       "25                             825  \n",
       "13                            1430  \n",
       "14                            1242  \n",
       "8                              104  \n",
       "7                             2776  \n",
       "23                            1062  \n",
       "21                            1486  \n",
       "6                             1043  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# format KMA-species results\n",
    "#\n",
    "\n",
    "# read in the results from KMA SPECIES-finding\n",
    "kma_species_output = pd.read_csv(f\"{output_dir}{prefix}.rgi.kma.kmerspecies_61mer_analysis.gene.txt\", sep='\\t')\n",
    "\n",
    "# convert 'ARO term' to lowercase for matching\n",
    "kma_species_output['ARO term'] = [i.lower() for i in kma_species_output['ARO term']] \n",
    "kma_species_output.sort_values(by='ARO term', inplace=True)\n",
    "kma_species_output.reset_index()\n",
    "\n",
    "# display subset of data for manual visual-checking\n",
    "kma_species_output[['ARO term', 'CARD*kmer Prediction', 'Mapped reads with kmer DB hits']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge Data!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID_contig_amr</th>\n",
       "      <th>Contig_contig_amr</th>\n",
       "      <th>Start_contig_amr</th>\n",
       "      <th>Stop_contig_amr</th>\n",
       "      <th>Orientation_contig_amr</th>\n",
       "      <th>Cut_Off_contig_amr</th>\n",
       "      <th>Pass_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_ARO_contig_amr</th>\n",
       "      <th>Best_Identities_contig_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Model_ID_contig_amr</th>\n",
       "      <th>Nudged_contig_amr</th>\n",
       "      <th>Note_contig_amr</th>\n",
       "      <th>ORF_ID_contig_sp</th>\n",
       "      <th>Contig_contig_sp</th>\n",
       "      <th>Cut_Off_contig_sp</th>\n",
       "      <th>Best_Hit_ARO_contig_sp</th>\n",
       "      <th>CARD*kmer Prediction_contig_sp</th>\n",
       "      <th>Taxonomic kmers_contig_sp</th>\n",
       "      <th>Genomic kmers_contig_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9 # 11869 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9</td>\n",
       "      <td>11869</td>\n",
       "      <td>12528</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>410</td>\n",
       "      <td>439.884</td>\n",
       "      <td>arlr</td>\n",
       "      <td>99.54</td>\n",
       "      <td>...</td>\n",
       "      <td>1390</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9 # 11869 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9</td>\n",
       "      <td>Strict</td>\n",
       "      <td>arlr</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 73;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 539;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8 # 10517 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8</td>\n",
       "      <td>10517</td>\n",
       "      <td>11872</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>880</td>\n",
       "      <td>911.753</td>\n",
       "      <td>arls</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8 # 10517 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>arls</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 574;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1296;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78 # 69097 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78</td>\n",
       "      <td>69097</td>\n",
       "      <td>70452</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>850</td>\n",
       "      <td>885.174</td>\n",
       "      <td>mepa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78 # 69097 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepa</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 455;</td>\n",
       "      <td>chr + plasmid: 1296; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79 # 70559 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79</td>\n",
       "      <td>70559</td>\n",
       "      <td>70978</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>250</td>\n",
       "      <td>285.804</td>\n",
       "      <td>mepr</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79 # 70559 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mepr</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 318;</td>\n",
       "      <td>chr + plasmid: 360; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9 # 7790 # ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9</td>\n",
       "      <td>7790</td>\n",
       "      <td>8233</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>270</td>\n",
       "      <td>301.212</td>\n",
       "      <td>mgra</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9 # 7790 # ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>mgra</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 196;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 384;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>5162</td>\n",
       "      <td>6007</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>500</td>\n",
       "      <td>506.523</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>93.95</td>\n",
       "      <td>...</td>\n",
       "      <td>726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>Strict</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 170;</td>\n",
       "      <td>chr + plasmid: 770; plasmid: 16; chr: 0;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>1615</td>\n",
       "      <td>3057</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>800</td>\n",
       "      <td>956.051</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>3276</td>\n",
       "      <td>True</td>\n",
       "      <td>possible complete gene, reference contained wi...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 914;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1386;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>34753</td>\n",
       "      <td>36018</td>\n",
       "      <td>+</td>\n",
       "      <td>Strict</td>\n",
       "      <td>800</td>\n",
       "      <td>840.877</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>99.29</td>\n",
       "      <td>...</td>\n",
       "      <td>2285</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 1206;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1206;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>15884</td>\n",
       "      <td>17050</td>\n",
       "      <td>+</td>\n",
       "      <td>Strict</td>\n",
       "      <td>700</td>\n",
       "      <td>763.066</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>99.74</td>\n",
       "      <td>...</td>\n",
       "      <td>3367</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 656;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1107;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>157</td>\n",
       "      <td>1515</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>850</td>\n",
       "      <td>871.692</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>99.78</td>\n",
       "      <td>...</td>\n",
       "      <td>1835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>Strict</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 1109;</td>\n",
       "      <td>chr + plasmid: 1299; plasmid: 0; chr: 0;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows  32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   ORF_ID_contig_amr  \\\n",
       "0  NODE_47_length_13669_cov_51.565374_9 # 11869 #...   \n",
       "1  NODE_47_length_13669_cov_51.565374_8 # 10517 #...   \n",
       "2  NODE_4_length_124562_cov_63.896729_78 # 69097 ...   \n",
       "3  NODE_4_length_124562_cov_63.896729_79 # 70559 ...   \n",
       "4  NODE_41_length_18243_cov_57.681328_9 # 7790 # ...   \n",
       "5  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "6  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "7  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "8  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "9  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                       Contig_contig_amr  Start_contig_amr  Stop_contig_amr  \\\n",
       "0   NODE_47_length_13669_cov_51.565374_9             11869            12528   \n",
       "1   NODE_47_length_13669_cov_51.565374_8             10517            11872   \n",
       "2  NODE_4_length_124562_cov_63.896729_78             69097            70452   \n",
       "3  NODE_4_length_124562_cov_63.896729_79             70559            70978   \n",
       "4   NODE_41_length_18243_cov_57.681328_9              7790             8233   \n",
       "5  NODE_46_length_14877_cov_127.795875_5              5162             6007   \n",
       "6   NODE_37_length_19743_cov_60.568519_2              1615             3057   \n",
       "7  NODE_13_length_74983_cov_65.352979_38             34753            36018   \n",
       "8  NODE_41_length_18243_cov_57.681328_17             15884            17050   \n",
       "9   NODE_32_length_28365_cov_66.015789_2               157             1515   \n",
       "\n",
       "  Orientation_contig_amr Cut_Off_contig_amr  Pass_Bitscore_contig_amr  \\\n",
       "0                      -             Strict                       410   \n",
       "1                      -            Perfect                       880   \n",
       "2                      -            Perfect                       850   \n",
       "3                      -            Perfect                       250   \n",
       "4                      -            Perfect                       270   \n",
       "5                      -             Strict                       500   \n",
       "6                      -            Perfect                       800   \n",
       "7                      +             Strict                       800   \n",
       "8                      +             Strict                       700   \n",
       "9                      -             Strict                       850   \n",
       "\n",
       "   Best_Hit_Bitscore_contig_amr  \\\n",
       "0                       439.884   \n",
       "1                       911.753   \n",
       "2                       885.174   \n",
       "3                       285.804   \n",
       "4                       301.212   \n",
       "5                       506.523   \n",
       "6                       956.051   \n",
       "7                       840.877   \n",
       "8                       763.066   \n",
       "9                       871.692   \n",
       "\n",
       "                             Best_Hit_ARO_contig_amr  \\\n",
       "0                                               arlr   \n",
       "1                                               arls   \n",
       "2                                               mepa   \n",
       "3                                               mepr   \n",
       "4                                               mgra   \n",
       "5                          pc1 beta-lactamase (blaz)   \n",
       "6                         staphylococcus aureus lmrs   \n",
       "7  staphylococcus aureus mura with mutation confe...   \n",
       "8                         staphylococcus aureus nora   \n",
       "9                                            tet(38)   \n",
       "\n",
       "   Best_Identities_contig_amr  ...  Model_ID_contig_amr Nudged_contig_amr  \\\n",
       "0                       99.54  ...                 1390               NaN   \n",
       "1                      100.00  ...                  182               NaN   \n",
       "2                      100.00  ...                  377               NaN   \n",
       "3                      100.00  ...                  856               NaN   \n",
       "4                      100.00  ...                  154               NaN   \n",
       "5                       93.95  ...                  726               NaN   \n",
       "6                      100.00  ...                 3276              True   \n",
       "7                       99.29  ...                 2285               NaN   \n",
       "8                       99.74  ...                 3367               NaN   \n",
       "9                       99.78  ...                 1835               NaN   \n",
       "\n",
       "                                     Note_contig_amr  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4                                                NaN   \n",
       "5                                                NaN   \n",
       "6  possible complete gene, reference contained wi...   \n",
       "7                                                NaN   \n",
       "8                                                NaN   \n",
       "9                                                NaN   \n",
       "\n",
       "                                    ORF_ID_contig_sp  \\\n",
       "0  NODE_47_length_13669_cov_51.565374_9 # 11869 #...   \n",
       "1  NODE_47_length_13669_cov_51.565374_8 # 10517 #...   \n",
       "2  NODE_4_length_124562_cov_63.896729_78 # 69097 ...   \n",
       "3  NODE_4_length_124562_cov_63.896729_79 # 70559 ...   \n",
       "4  NODE_41_length_18243_cov_57.681328_9 # 7790 # ...   \n",
       "5  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "6  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "7  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "8  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "9  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                        Contig_contig_sp Cut_Off_contig_sp  \\\n",
       "0   NODE_47_length_13669_cov_51.565374_9            Strict   \n",
       "1   NODE_47_length_13669_cov_51.565374_8           Perfect   \n",
       "2  NODE_4_length_124562_cov_63.896729_78           Perfect   \n",
       "3  NODE_4_length_124562_cov_63.896729_79           Perfect   \n",
       "4   NODE_41_length_18243_cov_57.681328_9           Perfect   \n",
       "5  NODE_46_length_14877_cov_127.795875_5            Strict   \n",
       "6   NODE_37_length_19743_cov_60.568519_2           Perfect   \n",
       "7  NODE_13_length_74983_cov_65.352979_38            Strict   \n",
       "8  NODE_41_length_18243_cov_57.681328_17            Strict   \n",
       "9   NODE_32_length_28365_cov_66.015789_2            Strict   \n",
       "\n",
       "                              Best_Hit_ARO_contig_sp  \\\n",
       "0                                               arlr   \n",
       "1                                               arls   \n",
       "2                                               mepa   \n",
       "3                                               mepr   \n",
       "4                                               mgra   \n",
       "5                          pc1 beta-lactamase (blaz)   \n",
       "6                         staphylococcus aureus lmrs   \n",
       "7  staphylococcus aureus mura with mutation confe...   \n",
       "8                         staphylococcus aureus nora   \n",
       "9                                            tet(38)   \n",
       "\n",
       "                  CARD*kmer Prediction_contig_sp  \\\n",
       "0             Staphylococcus aureus (chromosome)   \n",
       "1             Staphylococcus aureus (chromosome)   \n",
       "2  Staphylococcus aureus (chromosome or plasmid)   \n",
       "3  Staphylococcus aureus (chromosome or plasmid)   \n",
       "4             Staphylococcus aureus (chromosome)   \n",
       "5  Staphylococcus aureus (chromosome or plasmid)   \n",
       "6             Staphylococcus aureus (chromosome)   \n",
       "7             Staphylococcus aureus (chromosome)   \n",
       "8             Staphylococcus aureus (chromosome)   \n",
       "9  Staphylococcus aureus (chromosome or plasmid)   \n",
       "\n",
       "       Taxonomic kmers_contig_sp                    Genomic kmers_contig_sp  \n",
       "0    Staphylococcus aureus: 73;    chr + plasmid: 0; plasmid: 0; chr: 539;   \n",
       "1   Staphylococcus aureus: 574;   chr + plasmid: 0; plasmid: 0; chr: 1296;   \n",
       "2   Staphylococcus aureus: 455;   chr + plasmid: 1296; plasmid: 0; chr: 0;   \n",
       "3   Staphylococcus aureus: 318;    chr + plasmid: 360; plasmid: 0; chr: 0;   \n",
       "4   Staphylococcus aureus: 196;    chr + plasmid: 0; plasmid: 0; chr: 384;   \n",
       "5   Staphylococcus aureus: 170;   chr + plasmid: 770; plasmid: 16; chr: 0;   \n",
       "6   Staphylococcus aureus: 914;   chr + plasmid: 0; plasmid: 0; chr: 1386;   \n",
       "7  Staphylococcus aureus: 1206;   chr + plasmid: 0; plasmid: 0; chr: 1206;   \n",
       "8   Staphylococcus aureus: 656;   chr + plasmid: 0; plasmid: 0; chr: 1107;   \n",
       "9  Staphylococcus aureus: 1109;   chr + plasmid: 1299; plasmid: 0; chr: 0;   \n",
       "\n",
       "[10 rows x 32 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# merge the main-amr to main-species results\n",
    "#\n",
    "\n",
    "# rename main-amr and main-species columns to account for duplciate column names\n",
    "main_output.columns = [i+'_contig_amr' for i in main_output.columns]\n",
    "main_species_output.columns = [i+'_contig_sp' for i in main_species_output.columns]\n",
    "\n",
    "# vv get rid of weird additional space in some of the contig amr fields to make matching work downstream\n",
    "main_output['Contig_contig_amr'] = [i.strip() for i in main_output['Contig_contig_amr']] \n",
    "\n",
    "# merge the data where Best_Hit_ARO and Contig name must match\n",
    "merge_a = main_output.merge(main_species_output, left_on = ['Best_Hit_ARO_contig_amr', 'Contig_contig_amr'],\n",
    "                                                               right_on = ['Best_Hit_ARO_contig_sp', 'Contig_contig_sp'], how='outer',\n",
    "                                                              suffixes = [None, None])\n",
    "merge_a.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID_contig_amr</th>\n",
       "      <th>Contig_contig_amr</th>\n",
       "      <th>Start_contig_amr</th>\n",
       "      <th>Stop_contig_amr</th>\n",
       "      <th>Orientation_contig_amr</th>\n",
       "      <th>Cut_Off_contig_amr</th>\n",
       "      <th>Pass_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_ARO_contig_amr</th>\n",
       "      <th>Best_Identities_contig_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Nudged_contig_amr</th>\n",
       "      <th>Note_contig_amr</th>\n",
       "      <th>ORF_ID_contig_sp</th>\n",
       "      <th>Contig_contig_sp</th>\n",
       "      <th>Cut_Off_contig_sp</th>\n",
       "      <th>Best_Hit_ARO_contig_sp</th>\n",
       "      <th>CARD*kmer Prediction_contig_sp</th>\n",
       "      <th>Taxonomic kmers_contig_sp</th>\n",
       "      <th>Genomic kmers_contig_sp</th>\n",
       "      <th>ARO_contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>5162</td>\n",
       "      <td>6007</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>500</td>\n",
       "      <td>506.523</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>93.95</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5 # 5162 #...</td>\n",
       "      <td>NODE_46_length_14877_cov_127.795875_5</td>\n",
       "      <td>Strict</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 170;</td>\n",
       "      <td>chr + plasmid: 770; plasmid: 16; chr: 0;</td>\n",
       "      <td>pc1 beta-lactamase (blaz)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>1615</td>\n",
       "      <td>3057</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>800</td>\n",
       "      <td>956.051</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>True</td>\n",
       "      <td>possible complete gene, reference contained wi...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2 # 1615 # ...</td>\n",
       "      <td>NODE_37_length_19743_cov_60.568519_2</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 914;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1386;</td>\n",
       "      <td>staphylococcus aureus lmrs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>34753</td>\n",
       "      <td>36018</td>\n",
       "      <td>+</td>\n",
       "      <td>Strict</td>\n",
       "      <td>800</td>\n",
       "      <td>840.877</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>99.29</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38 # 34753 ...</td>\n",
       "      <td>NODE_13_length_74983_cov_65.352979_38</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 1206;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1206;</td>\n",
       "      <td>staphylococcus aureus mura with mutation confe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>15884</td>\n",
       "      <td>17050</td>\n",
       "      <td>+</td>\n",
       "      <td>Strict</td>\n",
       "      <td>700</td>\n",
       "      <td>763.066</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>99.74</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>Strict</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>Staphylococcus aureus (chromosome)</td>\n",
       "      <td>Staphylococcus aureus: 656;</td>\n",
       "      <td>chr + plasmid: 0; plasmid: 0; chr: 1107;</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>157</td>\n",
       "      <td>1515</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>850</td>\n",
       "      <td>871.692</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>99.78</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>Strict</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>Staphylococcus aureus (chromosome or plasmid)</td>\n",
       "      <td>Staphylococcus aureus: 1109;</td>\n",
       "      <td>chr + plasmid: 1299; plasmid: 0; chr: 0;</td>\n",
       "      <td>tet(38)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   ORF_ID_contig_amr  \\\n",
       "5  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "6  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "7  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "8  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "9  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                       Contig_contig_amr  Start_contig_amr  Stop_contig_amr  \\\n",
       "5  NODE_46_length_14877_cov_127.795875_5              5162             6007   \n",
       "6   NODE_37_length_19743_cov_60.568519_2              1615             3057   \n",
       "7  NODE_13_length_74983_cov_65.352979_38             34753            36018   \n",
       "8  NODE_41_length_18243_cov_57.681328_17             15884            17050   \n",
       "9   NODE_32_length_28365_cov_66.015789_2               157             1515   \n",
       "\n",
       "  Orientation_contig_amr Cut_Off_contig_amr  Pass_Bitscore_contig_amr  \\\n",
       "5                      -             Strict                       500   \n",
       "6                      -            Perfect                       800   \n",
       "7                      +             Strict                       800   \n",
       "8                      +             Strict                       700   \n",
       "9                      -             Strict                       850   \n",
       "\n",
       "   Best_Hit_Bitscore_contig_amr  \\\n",
       "5                       506.523   \n",
       "6                       956.051   \n",
       "7                       840.877   \n",
       "8                       763.066   \n",
       "9                       871.692   \n",
       "\n",
       "                             Best_Hit_ARO_contig_amr  \\\n",
       "5                          pc1 beta-lactamase (blaz)   \n",
       "6                         staphylococcus aureus lmrs   \n",
       "7  staphylococcus aureus mura with mutation confe...   \n",
       "8                         staphylococcus aureus nora   \n",
       "9                                            tet(38)   \n",
       "\n",
       "   Best_Identities_contig_amr  ...  Nudged_contig_amr  \\\n",
       "5                       93.95  ...                NaN   \n",
       "6                      100.00  ...               True   \n",
       "7                       99.29  ...                NaN   \n",
       "8                       99.74  ...                NaN   \n",
       "9                       99.78  ...                NaN   \n",
       "\n",
       "                                     Note_contig_amr  \\\n",
       "5                                                NaN   \n",
       "6  possible complete gene, reference contained wi...   \n",
       "7                                                NaN   \n",
       "8                                                NaN   \n",
       "9                                                NaN   \n",
       "\n",
       "                                    ORF_ID_contig_sp  \\\n",
       "5  NODE_46_length_14877_cov_127.795875_5 # 5162 #...   \n",
       "6  NODE_37_length_19743_cov_60.568519_2 # 1615 # ...   \n",
       "7  NODE_13_length_74983_cov_65.352979_38 # 34753 ...   \n",
       "8  NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "9  NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "\n",
       "                        Contig_contig_sp Cut_Off_contig_sp  \\\n",
       "5  NODE_46_length_14877_cov_127.795875_5            Strict   \n",
       "6   NODE_37_length_19743_cov_60.568519_2           Perfect   \n",
       "7  NODE_13_length_74983_cov_65.352979_38            Strict   \n",
       "8  NODE_41_length_18243_cov_57.681328_17            Strict   \n",
       "9   NODE_32_length_28365_cov_66.015789_2            Strict   \n",
       "\n",
       "                              Best_Hit_ARO_contig_sp  \\\n",
       "5                          pc1 beta-lactamase (blaz)   \n",
       "6                         staphylococcus aureus lmrs   \n",
       "7  staphylococcus aureus mura with mutation confe...   \n",
       "8                         staphylococcus aureus nora   \n",
       "9                                            tet(38)   \n",
       "\n",
       "                  CARD*kmer Prediction_contig_sp  \\\n",
       "5  Staphylococcus aureus (chromosome or plasmid)   \n",
       "6             Staphylococcus aureus (chromosome)   \n",
       "7             Staphylococcus aureus (chromosome)   \n",
       "8             Staphylococcus aureus (chromosome)   \n",
       "9  Staphylococcus aureus (chromosome or plasmid)   \n",
       "\n",
       "       Taxonomic kmers_contig_sp                    Genomic kmers_contig_sp  \\\n",
       "5   Staphylococcus aureus: 170;   chr + plasmid: 770; plasmid: 16; chr: 0;    \n",
       "6   Staphylococcus aureus: 914;   chr + plasmid: 0; plasmid: 0; chr: 1386;    \n",
       "7  Staphylococcus aureus: 1206;   chr + plasmid: 0; plasmid: 0; chr: 1206;    \n",
       "8   Staphylococcus aureus: 656;   chr + plasmid: 0; plasmid: 0; chr: 1107;    \n",
       "9  Staphylococcus aureus: 1109;   chr + plasmid: 1299; plasmid: 0; chr: 0;    \n",
       "\n",
       "                                          ARO_contig  \n",
       "5                          pc1 beta-lactamase (blaz)  \n",
       "6                         staphylococcus aureus lmrs  \n",
       "7  staphylococcus aureus mura with mutation confe...  \n",
       "8                         staphylococcus aureus nora  \n",
       "9                                            tet(38)  \n",
       "\n",
       "[5 rows x 33 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dealing with NAs where the gene name doesn't match between AMR calling and species calling\n",
    "# the NAs cause downstream issues with the merging!!\n",
    "merge_a['ARO_contig'] = [merge_a.iloc[i]['Best_Hit_ARO_contig_amr'] if str(merge_a.iloc[i]['Best_Hit_ARO_contig_amr']) != 'nan' else merge_a.iloc[i]['Best_Hit_ARO_contig_sp'] for i in range(len(merge_a.index))]\n",
    "merge_a.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Reference Sequence_kma_amr</th>\n",
       "      <th>ARO Term_kma_amr</th>\n",
       "      <th>ARO Accession_kma_amr</th>\n",
       "      <th>Reference Model Type_kma_amr</th>\n",
       "      <th>Reference DB_kma_amr</th>\n",
       "      <th>Reference Allele Source_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed in Genome(s)_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed in Plasmid(s)_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed Pathogen(s)_kma_amr</th>\n",
       "      <th>Completely Mapped Reads_kma_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Single species (plasmid) reads_kma_sp</th>\n",
       "      <th>Single species (no genomic info) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (no genomic info) reads_kma_sp</th>\n",
       "      <th>Promiscuous plasmid reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (no genomic info) reads_kma_sp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARO:3000838|ID:1390|Name:arlR|NCBI:CP023390.1</td>\n",
       "      <td>arlr</td>\n",
       "      <td>3000838.0</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>CARD</td>\n",
       "      <td>CARD curation</td>\n",
       "      <td>no data</td>\n",
       "      <td>no data</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>397.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Staphylococcus: 4;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>207.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ARO:3000839|ID:182|Name:arlS|NCBI:CP000253.1</td>\n",
       "      <td>arls</td>\n",
       "      <td>3000839.0</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>CARD</td>\n",
       "      <td>CARD curation</td>\n",
       "      <td>no data</td>\n",
       "      <td>no data</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>791.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Staphylococcus: 26;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>304.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARO:3004146|ID:2800|Name:cfrC|NCBI:AM412317.1</td>\n",
       "      <td>cfrc</td>\n",
       "      <td>3004146.0</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>CARD</td>\n",
       "      <td>CARD curation</td>\n",
       "      <td>no data</td>\n",
       "      <td>no data</td>\n",
       "      <td>Clostridium botulinum</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ARO:3005097|ID:3829|Name:mecC-type_BlaZ|NCBI:F...</td>\n",
       "      <td>mecc-type blaz</td>\n",
       "      <td>3005097.0</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>CARD</td>\n",
       "      <td>CARD curation</td>\n",
       "      <td>no data</td>\n",
       "      <td>no data</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>477.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ARO:3000026|ID:377|Name:mepA|NCBI:AY661734.1</td>\n",
       "      <td>mepa</td>\n",
       "      <td>3000026.0</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>CARD</td>\n",
       "      <td>CARD curation</td>\n",
       "      <td>no data</td>\n",
       "      <td>no data</td>\n",
       "      <td>Staphylococcus aureus</td>\n",
       "      <td>1202.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>426.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Reference Sequence_kma_amr ARO Term_kma_amr  \\\n",
       "0       ARO:3000838|ID:1390|Name:arlR|NCBI:CP023390.1             arlr   \n",
       "1        ARO:3000839|ID:182|Name:arlS|NCBI:CP000253.1             arls   \n",
       "3       ARO:3004146|ID:2800|Name:cfrC|NCBI:AM412317.1             cfrc   \n",
       "9   ARO:3005097|ID:3829|Name:mecC-type_BlaZ|NCBI:F...   mecc-type blaz   \n",
       "10       ARO:3000026|ID:377|Name:mepA|NCBI:AY661734.1             mepa   \n",
       "\n",
       "    ARO Accession_kma_amr Reference Model Type_kma_amr Reference DB_kma_amr  \\\n",
       "0               3000838.0        protein homolog model                 CARD   \n",
       "1               3000839.0        protein homolog model                 CARD   \n",
       "3               3004146.0        protein homolog model                 CARD   \n",
       "9               3005097.0        protein homolog model                 CARD   \n",
       "10              3000026.0        protein homolog model                 CARD   \n",
       "\n",
       "   Reference Allele Source_kma_amr  \\\n",
       "0                    CARD curation   \n",
       "1                    CARD curation   \n",
       "3                    CARD curation   \n",
       "9                    CARD curation   \n",
       "10                   CARD curation   \n",
       "\n",
       "   Resistomes & Variants: Observed in Genome(s)_kma_amr  \\\n",
       "0                                             no data     \n",
       "1                                             no data     \n",
       "3                                             no data     \n",
       "9                                             no data     \n",
       "10                                            no data     \n",
       "\n",
       "   Resistomes & Variants: Observed in Plasmid(s)_kma_amr  \\\n",
       "0                                             no data      \n",
       "1                                             no data      \n",
       "3                                             no data      \n",
       "9                                             no data      \n",
       "10                                            no data      \n",
       "\n",
       "   Resistomes & Variants: Observed Pathogen(s)_kma_amr  \\\n",
       "0                               Staphylococcus aureus    \n",
       "1                               Staphylococcus aureus    \n",
       "3                               Clostridium botulinum    \n",
       "9                               Staphylococcus aureus    \n",
       "10                              Staphylococcus aureus    \n",
       "\n",
       "    Completely Mapped Reads_kma_amr  ...  \\\n",
       "0                             397.0  ...   \n",
       "1                             791.0  ...   \n",
       "3                               1.0  ...   \n",
       "9                             477.0  ...   \n",
       "10                           1202.0  ...   \n",
       "\n",
       "    Single species (plasmid) reads_kma_sp  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "3                                     NaN   \n",
       "9                                     NaN   \n",
       "10                                    NaN   \n",
       "\n",
       "    Single species (no genomic info) reads_kma_sp  \\\n",
       "0                                             NaN   \n",
       "1                                             NaN   \n",
       "3                                             NaN   \n",
       "9                                             NaN   \n",
       "10                                            NaN   \n",
       "\n",
       "    Single genus (chromosome) reads_kma_sp  \\\n",
       "0                      Staphylococcus: 4;    \n",
       "1                     Staphylococcus: 26;    \n",
       "3                                      NaN   \n",
       "9                                      NaN   \n",
       "10                                     NaN   \n",
       "\n",
       "    Single genus (chromosome or plasmid) reads_kma_sp  \\\n",
       "0                                                 NaN   \n",
       "1                                                 NaN   \n",
       "3                                                 NaN   \n",
       "9                                                 NaN   \n",
       "10                                                NaN   \n",
       "\n",
       "    Single genus (plasmid) reads_kma_sp  \\\n",
       "0                                   NaN   \n",
       "1                                   NaN   \n",
       "3                                   NaN   \n",
       "9                                   NaN   \n",
       "10                                  NaN   \n",
       "\n",
       "   Single genus (no genomic info) reads_kma_sp  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "3                                          NaN   \n",
       "9                                          NaN   \n",
       "10                                         NaN   \n",
       "\n",
       "    Promiscuous plasmid reads_kma_sp  \\\n",
       "0                                0.0   \n",
       "1                                0.0   \n",
       "3                                NaN   \n",
       "9                                NaN   \n",
       "10                               0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome) reads_kma_sp  \\\n",
       "0                                       207.0   \n",
       "1                                       304.0   \n",
       "3                                         NaN   \n",
       "9                                         NaN   \n",
       "10                                        0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome or plasmid) reads_kma_sp  \\\n",
       "0                                                 0.0      \n",
       "1                                                 0.0      \n",
       "3                                                 NaN      \n",
       "9                                                 NaN      \n",
       "10                                              426.0      \n",
       "\n",
       "   Unknown taxonomy (no genomic info) reads_kma_sp  \n",
       "0                                              0.0  \n",
       "1                                              0.0  \n",
       "3                                              NaN  \n",
       "9                                              NaN  \n",
       "10                                             0.0  \n",
       "\n",
       "[5 rows x 39 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "# merge kma-amr kma-species results\n",
    "#\n",
    "\n",
    "#rename kma-amr and kma-species columns to account for duplicate column names\n",
    "kma_output.columns = [i+'_kma_amr' for i in kma_output.columns]  # ARO Term\n",
    "kma_species_output.columns = [i+'_kma_sp' for i in kma_species_output.columns]  #ARO term\n",
    "merge_b = kma_output.merge(kma_species_output, left_on = 'ARO Term_kma_amr', right_on = 'ARO term_kma_sp',\n",
    "                           how = 'outer', suffixes = [None, None])\n",
    "\n",
    "# remove kma results from variant models (because these results are inaccurate)\n",
    "merge_b = merge_b[merge_b['Reference Model Type_kma_amr'] != \"protein variant model\"] # remove protein variant model\n",
    "merge_b = merge_b[merge_b['Reference Model Type_kma_amr'] != \"rRNA gene variant model\"] # remove rRNA variant model\n",
    "\n",
    "merge_b.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Reference Sequence_kma_amr</th>\n",
       "      <th>ARO Term_kma_amr</th>\n",
       "      <th>ARO Accession_kma_amr</th>\n",
       "      <th>Reference Model Type_kma_amr</th>\n",
       "      <th>Reference DB_kma_amr</th>\n",
       "      <th>Reference Allele Source_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed in Genome(s)_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed in Plasmid(s)_kma_amr</th>\n",
       "      <th>Resistomes &amp; Variants: Observed Pathogen(s)_kma_amr</th>\n",
       "      <th>Completely Mapped Reads_kma_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Single species (no genomic info) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (no genomic info) reads_kma_sp</th>\n",
       "      <th>Promiscuous plasmid reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (no genomic info) reads_kma_sp</th>\n",
       "      <th>ARO_kma</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna (rrsb) mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus fusa with mutation confe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus mprf with mutation confe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Reference Sequence_kma_amr ARO Term_kma_amr  ARO Accession_kma_amr  \\\n",
       "41                        NaN              NaN                    NaN   \n",
       "42                        NaN              NaN                    NaN   \n",
       "43                        NaN              NaN                    NaN   \n",
       "44                        NaN              NaN                    NaN   \n",
       "45                        NaN              NaN                    NaN   \n",
       "\n",
       "   Reference Model Type_kma_amr Reference DB_kma_amr  \\\n",
       "41                          NaN                  NaN   \n",
       "42                          NaN                  NaN   \n",
       "43                          NaN                  NaN   \n",
       "44                          NaN                  NaN   \n",
       "45                          NaN                  NaN   \n",
       "\n",
       "   Reference Allele Source_kma_amr  \\\n",
       "41                             NaN   \n",
       "42                             NaN   \n",
       "43                             NaN   \n",
       "44                             NaN   \n",
       "45                             NaN   \n",
       "\n",
       "   Resistomes & Variants: Observed in Genome(s)_kma_amr  \\\n",
       "41                                                NaN     \n",
       "42                                                NaN     \n",
       "43                                                NaN     \n",
       "44                                                NaN     \n",
       "45                                                NaN     \n",
       "\n",
       "   Resistomes & Variants: Observed in Plasmid(s)_kma_amr  \\\n",
       "41                                                NaN      \n",
       "42                                                NaN      \n",
       "43                                                NaN      \n",
       "44                                                NaN      \n",
       "45                                                NaN      \n",
       "\n",
       "   Resistomes & Variants: Observed Pathogen(s)_kma_amr  \\\n",
       "41                                                NaN    \n",
       "42                                                NaN    \n",
       "43                                                NaN    \n",
       "44                                                NaN    \n",
       "45                                                NaN    \n",
       "\n",
       "    Completely Mapped Reads_kma_amr  ...  \\\n",
       "41                              NaN  ...   \n",
       "42                              NaN  ...   \n",
       "43                              NaN  ...   \n",
       "44                              NaN  ...   \n",
       "45                              NaN  ...   \n",
       "\n",
       "    Single species (no genomic info) reads_kma_sp  \\\n",
       "41                                            NaN   \n",
       "42                                            NaN   \n",
       "43                                            NaN   \n",
       "44                                            NaN   \n",
       "45                                            NaN   \n",
       "\n",
       "    Single genus (chromosome) reads_kma_sp  \\\n",
       "41                                     NaN   \n",
       "42                                     NaN   \n",
       "43                                     NaN   \n",
       "44                                     NaN   \n",
       "45                                     NaN   \n",
       "\n",
       "    Single genus (chromosome or plasmid) reads_kma_sp  \\\n",
       "41                                                NaN   \n",
       "42                                                NaN   \n",
       "43                                                NaN   \n",
       "44                                                NaN   \n",
       "45                                                NaN   \n",
       "\n",
       "    Single genus (plasmid) reads_kma_sp  \\\n",
       "41                                  NaN   \n",
       "42                                  NaN   \n",
       "43                                  NaN   \n",
       "44                                  NaN   \n",
       "45                                  NaN   \n",
       "\n",
       "    Single genus (no genomic info) reads_kma_sp  \\\n",
       "41                                          NaN   \n",
       "42                                          NaN   \n",
       "43                                          NaN   \n",
       "44                                          NaN   \n",
       "45                                          NaN   \n",
       "\n",
       "   Promiscuous plasmid reads_kma_sp  \\\n",
       "41                              0.0   \n",
       "42                              0.0   \n",
       "43                              0.0   \n",
       "44                              0.0   \n",
       "45                              0.0   \n",
       "\n",
       "    Unknown taxonomy (chromosome) reads_kma_sp  \\\n",
       "41                                         0.0   \n",
       "42                                         0.0   \n",
       "43                                         0.0   \n",
       "44                                       146.0   \n",
       "45                                         0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome or plasmid) reads_kma_sp  \\\n",
       "41                                                0.0      \n",
       "42                                                0.0      \n",
       "43                                                0.0      \n",
       "44                                                0.0      \n",
       "45                                                0.0      \n",
       "\n",
       "   Unknown taxonomy (no genomic info) reads_kma_sp  \\\n",
       "41                                             0.0   \n",
       "42                                             0.0   \n",
       "43                                             0.0   \n",
       "44                                             0.0   \n",
       "45                                             0.0   \n",
       "\n",
       "                                              ARO_kma  \n",
       "41  mycolicibacterium smegmatis 16s rrna (rrsb) mu...  \n",
       "42  staphylococcus aureus fusa with mutation confe...  \n",
       "43  staphylococcus aureus mprf with mutation confe...  \n",
       "44  staphylococcus aureus parc conferring resistan...  \n",
       "45  staphylococcus aureus walk with mutation confe...  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dealing with NAs where the gene name doesn't match between AMR calling and species calling\n",
    "# the NAs cause downstream issues with the merging!!\n",
    "merge_b['ARO_kma'] = [merge_b.iloc[i]['ARO Term_kma_amr'] if str(merge_b.iloc[i]['ARO Term_kma_amr']) != 'nan' else merge_b.iloc[i]['ARO term_kma_sp'] for i in range(len(merge_b.index))]\n",
    "merge_b.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID_contig_amr</th>\n",
       "      <th>Contig_contig_amr</th>\n",
       "      <th>Start_contig_amr</th>\n",
       "      <th>Stop_contig_amr</th>\n",
       "      <th>Orientation_contig_amr</th>\n",
       "      <th>Cut_Off_contig_amr</th>\n",
       "      <th>Pass_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_ARO_contig_amr</th>\n",
       "      <th>Best_Identities_contig_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Single species (no genomic info) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome) reads_kma_sp</th>\n",
       "      <th>Single genus (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (plasmid) reads_kma_sp</th>\n",
       "      <th>Single genus (no genomic info) reads_kma_sp</th>\n",
       "      <th>Promiscuous plasmid reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (no genomic info) reads_kma_sp</th>\n",
       "      <th>ARO_kma</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna (rrsb) mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus fusa with mutation confe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus mprf with mutation confe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  73 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ORF_ID_contig_amr Contig_contig_amr  Start_contig_amr  Stop_contig_amr  \\\n",
       "13               NaN               NaN               NaN              NaN   \n",
       "14               NaN               NaN               NaN              NaN   \n",
       "15               NaN               NaN               NaN              NaN   \n",
       "16               NaN               NaN               NaN              NaN   \n",
       "17               NaN               NaN               NaN              NaN   \n",
       "\n",
       "   Orientation_contig_amr Cut_Off_contig_amr  Pass_Bitscore_contig_amr  \\\n",
       "13                    NaN                NaN                       NaN   \n",
       "14                    NaN                NaN                       NaN   \n",
       "15                    NaN                NaN                       NaN   \n",
       "16                    NaN                NaN                       NaN   \n",
       "17                    NaN                NaN                       NaN   \n",
       "\n",
       "    Best_Hit_Bitscore_contig_amr Best_Hit_ARO_contig_amr  \\\n",
       "13                           NaN                     NaN   \n",
       "14                           NaN                     NaN   \n",
       "15                           NaN                     NaN   \n",
       "16                           NaN                     NaN   \n",
       "17                           NaN                     NaN   \n",
       "\n",
       "    Best_Identities_contig_amr  ...  \\\n",
       "13                         NaN  ...   \n",
       "14                         NaN  ...   \n",
       "15                         NaN  ...   \n",
       "16                         NaN  ...   \n",
       "17                         NaN  ...   \n",
       "\n",
       "    Single species (no genomic info) reads_kma_sp  \\\n",
       "13                                            NaN   \n",
       "14                                            NaN   \n",
       "15                                            NaN   \n",
       "16                                            NaN   \n",
       "17                                            NaN   \n",
       "\n",
       "   Single genus (chromosome) reads_kma_sp  \\\n",
       "13                                    NaN   \n",
       "14                                    NaN   \n",
       "15                                    NaN   \n",
       "16                                    NaN   \n",
       "17                                    NaN   \n",
       "\n",
       "   Single genus (chromosome or plasmid) reads_kma_sp  \\\n",
       "13                                               NaN   \n",
       "14                                               NaN   \n",
       "15                                               NaN   \n",
       "16                                               NaN   \n",
       "17                                               NaN   \n",
       "\n",
       "    Single genus (plasmid) reads_kma_sp  \\\n",
       "13                                  NaN   \n",
       "14                                  NaN   \n",
       "15                                  NaN   \n",
       "16                                  NaN   \n",
       "17                                  NaN   \n",
       "\n",
       "   Single genus (no genomic info) reads_kma_sp  \\\n",
       "13                                         NaN   \n",
       "14                                         NaN   \n",
       "15                                         NaN   \n",
       "16                                         NaN   \n",
       "17                                         NaN   \n",
       "\n",
       "   Promiscuous plasmid reads_kma_sp  \\\n",
       "13                              0.0   \n",
       "14                              0.0   \n",
       "15                              0.0   \n",
       "16                              0.0   \n",
       "17                              0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome) reads_kma_sp  \\\n",
       "13                                        0.0   \n",
       "14                                        0.0   \n",
       "15                                        0.0   \n",
       "16                                      146.0   \n",
       "17                                        0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome or plasmid) reads_kma_sp  \\\n",
       "13                                                0.0      \n",
       "14                                                0.0      \n",
       "15                                                0.0      \n",
       "16                                                0.0      \n",
       "17                                                0.0      \n",
       "\n",
       "   Unknown taxonomy (no genomic info) reads_kma_sp  \\\n",
       "13                                             0.0   \n",
       "14                                             0.0   \n",
       "15                                             0.0   \n",
       "16                                             0.0   \n",
       "17                                             0.0   \n",
       "\n",
       "                                              ARO_kma  \n",
       "13  mycolicibacterium smegmatis 16s rrna (rrsb) mu...  \n",
       "14  staphylococcus aureus fusa with mutation confe...  \n",
       "15  staphylococcus aureus mprf with mutation confe...  \n",
       "16  staphylococcus aureus parc conferring resistan...  \n",
       "17  staphylococcus aureus walk with mutation confe...  \n",
       "\n",
       "[5 rows x 73 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# final merge of MAIN and KMA combined results\n",
    "merge_x = merge_a.merge(merge_b, left_on = 'ARO_contig',\n",
    "                        right_on = 'ARO_kma', how='outer',\n",
    "                        suffixes = [None, None])\n",
    "merge_x.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID_contig_amr</th>\n",
       "      <th>Contig_contig_amr</th>\n",
       "      <th>Start_contig_amr</th>\n",
       "      <th>Stop_contig_amr</th>\n",
       "      <th>Orientation_contig_amr</th>\n",
       "      <th>Cut_Off_contig_amr</th>\n",
       "      <th>Pass_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_ARO_contig_amr</th>\n",
       "      <th>Best_Identities_contig_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Promiscuous plasmid reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (no genomic info) reads_kma_sp</th>\n",
       "      <th>ARO_kma</th>\n",
       "      <th>ARO_overall</th>\n",
       "      <th>Gene_Family_overall</th>\n",
       "      <th>Drug_Class_overall</th>\n",
       "      <th>model_overall</th>\n",
       "      <th>Resistance_Mechanism_overall</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17 # 15884 ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_17</td>\n",
       "      <td>15884.0</td>\n",
       "      <td>17050.0</td>\n",
       "      <td>+</td>\n",
       "      <td>Strict</td>\n",
       "      <td>700.0</td>\n",
       "      <td>763.066</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>99.74</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>177.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>staphylococcus aureus nora</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>fluoroquinolone antibiotic</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2 # 157 # 1...</td>\n",
       "      <td>NODE_32_length_28365_cov_66.015789_2</td>\n",
       "      <td>157.0</td>\n",
       "      <td>1515.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>850.0</td>\n",
       "      <td>871.692</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>99.78</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>tet(38)</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>tetracycline antibiotic</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cfrc</td>\n",
       "      <td>cfrc</td>\n",
       "      <td>Cfr 23S ribosomal RNA methyltransferase</td>\n",
       "      <td>lincosamide antibiotic; streptogramin antibiot...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic target alteration</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mecc-type blaz</td>\n",
       "      <td>mecc-type blaz</td>\n",
       "      <td>blaZ beta-lactamase</td>\n",
       "      <td>penam</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus fosb</td>\n",
       "      <td>staphylococcus aureus fosb</td>\n",
       "      <td>fosfomycin thiol transferase</td>\n",
       "      <td>fosfomycin</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic inactivation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna (rrsb) mu...</td>\n",
       "      <td>mycolicibacterium smegmatis 16s rrna (rrsb) mu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus fusa with mutation confe...</td>\n",
       "      <td>staphylococcus aureus fusa with mutation confe...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus mprf with mutation confe...</td>\n",
       "      <td>staphylococcus aureus mprf with mutation confe...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>146.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "      <td>staphylococcus aureus parc conferring resistan...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "      <td>staphylococcus aureus walk with mutation confe...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows  78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    ORF_ID_contig_amr  \\\n",
       "8   NODE_41_length_18243_cov_57.681328_17 # 15884 ...   \n",
       "9   NODE_32_length_28365_cov_66.015789_2 # 157 # 1...   \n",
       "10                                                NaN   \n",
       "11                                                NaN   \n",
       "12                                                NaN   \n",
       "13                                                NaN   \n",
       "14                                                NaN   \n",
       "15                                                NaN   \n",
       "16                                                NaN   \n",
       "17                                                NaN   \n",
       "\n",
       "                        Contig_contig_amr  Start_contig_amr  Stop_contig_amr  \\\n",
       "8   NODE_41_length_18243_cov_57.681328_17           15884.0          17050.0   \n",
       "9    NODE_32_length_28365_cov_66.015789_2             157.0           1515.0   \n",
       "10                                    NaN               NaN              NaN   \n",
       "11                                    NaN               NaN              NaN   \n",
       "12                                    NaN               NaN              NaN   \n",
       "13                                    NaN               NaN              NaN   \n",
       "14                                    NaN               NaN              NaN   \n",
       "15                                    NaN               NaN              NaN   \n",
       "16                                    NaN               NaN              NaN   \n",
       "17                                    NaN               NaN              NaN   \n",
       "\n",
       "   Orientation_contig_amr Cut_Off_contig_amr  Pass_Bitscore_contig_amr  \\\n",
       "8                       +             Strict                     700.0   \n",
       "9                       -             Strict                     850.0   \n",
       "10                    NaN                NaN                       NaN   \n",
       "11                    NaN                NaN                       NaN   \n",
       "12                    NaN                NaN                       NaN   \n",
       "13                    NaN                NaN                       NaN   \n",
       "14                    NaN                NaN                       NaN   \n",
       "15                    NaN                NaN                       NaN   \n",
       "16                    NaN                NaN                       NaN   \n",
       "17                    NaN                NaN                       NaN   \n",
       "\n",
       "    Best_Hit_Bitscore_contig_amr     Best_Hit_ARO_contig_amr  \\\n",
       "8                        763.066  staphylococcus aureus nora   \n",
       "9                        871.692                     tet(38)   \n",
       "10                           NaN                         NaN   \n",
       "11                           NaN                         NaN   \n",
       "12                           NaN                         NaN   \n",
       "13                           NaN                         NaN   \n",
       "14                           NaN                         NaN   \n",
       "15                           NaN                         NaN   \n",
       "16                           NaN                         NaN   \n",
       "17                           NaN                         NaN   \n",
       "\n",
       "    Best_Identities_contig_amr  ...  Promiscuous plasmid reads_kma_sp  \\\n",
       "8                        99.74  ...                               0.0   \n",
       "9                        99.78  ...                               0.0   \n",
       "10                         NaN  ...                               NaN   \n",
       "11                         NaN  ...                               NaN   \n",
       "12                         NaN  ...                               0.0   \n",
       "13                         NaN  ...                               0.0   \n",
       "14                         NaN  ...                               0.0   \n",
       "15                         NaN  ...                               0.0   \n",
       "16                         NaN  ...                               0.0   \n",
       "17                         NaN  ...                               0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome) reads_kma_sp  \\\n",
       "8                                       177.0   \n",
       "9                                         0.0   \n",
       "10                                        NaN   \n",
       "11                                        NaN   \n",
       "12                                        1.0   \n",
       "13                                        0.0   \n",
       "14                                        0.0   \n",
       "15                                        0.0   \n",
       "16                                      146.0   \n",
       "17                                        0.0   \n",
       "\n",
       "   Unknown taxonomy (chromosome or plasmid) reads_kma_sp  \\\n",
       "8                                                 0.0      \n",
       "9                                                36.0      \n",
       "10                                                NaN      \n",
       "11                                                NaN      \n",
       "12                                                0.0      \n",
       "13                                                0.0      \n",
       "14                                                0.0      \n",
       "15                                                0.0      \n",
       "16                                                0.0      \n",
       "17                                                0.0      \n",
       "\n",
       "    Unknown taxonomy (no genomic info) reads_kma_sp  \\\n",
       "8                                               0.0   \n",
       "9                                               0.0   \n",
       "10                                              NaN   \n",
       "11                                              NaN   \n",
       "12                                              0.0   \n",
       "13                                              0.0   \n",
       "14                                              0.0   \n",
       "15                                              0.0   \n",
       "16                                              0.0   \n",
       "17                                              0.0   \n",
       "\n",
       "                                              ARO_kma  \\\n",
       "8                          staphylococcus aureus nora   \n",
       "9                                             tet(38)   \n",
       "10                                               cfrc   \n",
       "11                                     mecc-type blaz   \n",
       "12                         staphylococcus aureus fosb   \n",
       "13  mycolicibacterium smegmatis 16s rrna (rrsb) mu...   \n",
       "14  staphylococcus aureus fusa with mutation confe...   \n",
       "15  staphylococcus aureus mprf with mutation confe...   \n",
       "16  staphylococcus aureus parc conferring resistan...   \n",
       "17  staphylococcus aureus walk with mutation confe...   \n",
       "\n",
       "                                          ARO_overall  \\\n",
       "8                          staphylococcus aureus nora   \n",
       "9                                             tet(38)   \n",
       "10                                               cfrc   \n",
       "11                                     mecc-type blaz   \n",
       "12                         staphylococcus aureus fosb   \n",
       "13  mycolicibacterium smegmatis 16s rrna (rrsb) mu...   \n",
       "14  staphylococcus aureus fusa with mutation confe...   \n",
       "15  staphylococcus aureus mprf with mutation confe...   \n",
       "16  staphylococcus aureus parc conferring resistan...   \n",
       "17  staphylococcus aureus walk with mutation confe...   \n",
       "\n",
       "                                  Gene_Family_overall  \\\n",
       "8   major facilitator superfamily (MFS) antibiotic...   \n",
       "9   major facilitator superfamily (MFS) antibiotic...   \n",
       "10            Cfr 23S ribosomal RNA methyltransferase   \n",
       "11                                blaZ beta-lactamase   \n",
       "12                       fosfomycin thiol transferase   \n",
       "13                                                NaN   \n",
       "14                                                NaN   \n",
       "15                                                NaN   \n",
       "16                                                NaN   \n",
       "17                                                NaN   \n",
       "\n",
       "                                   Drug_Class_overall          model_overall  \\\n",
       "8                          fluoroquinolone antibiotic  protein homolog model   \n",
       "9                             tetracycline antibiotic  protein homolog model   \n",
       "10  lincosamide antibiotic; streptogramin antibiot...  protein homolog model   \n",
       "11                                              penam  protein homolog model   \n",
       "12                                         fosfomycin  protein homolog model   \n",
       "13                                                NaN                    NaN   \n",
       "14                                                NaN                    NaN   \n",
       "15                                                NaN                    NaN   \n",
       "16                                                NaN                    NaN   \n",
       "17                                                NaN                    NaN   \n",
       "\n",
       "    Resistance_Mechanism_overall  \n",
       "8              antibiotic efflux  \n",
       "9              antibiotic efflux  \n",
       "10  antibiotic target alteration  \n",
       "11       antibiotic inactivation  \n",
       "12       antibiotic inactivation  \n",
       "13                           NaN  \n",
       "14                           NaN  \n",
       "15                           NaN  \n",
       "16                           NaN  \n",
       "17                           NaN  \n",
       "\n",
       "[10 rows x 78 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create new columns to represent the overall outcome -- fill with MAIN result if exists, otherwise KMA result\n",
    "merge_x['ARO_overall'] = [merge_x.iloc[i]['ARO_contig'] if str(merge_x.iloc[i]['ARO_contig']) != 'nan' else merge_x.iloc[i]['ARO_kma'] for i in range(len(merge_x.index))]\n",
    "merge_x['Gene_Family_overall'] = [merge_x.iloc[i]['AMR Gene Family_contig_amr'] if not pd.isna(merge_x.iloc[i]['AMR Gene Family_contig_amr']) else merge_x.iloc[i]['AMR Gene Family_kma_amr'] for i in range(len(merge_x.index))]\n",
    "merge_x['Drug_Class_overall'] = [merge_x.iloc[i]['Drug Class_contig_amr'] if not pd.isna(merge_x.iloc[i]['Drug Class_contig_amr']) else merge_x.iloc[i]['Drug Class_kma_amr'] for i in range(len(merge_x.index))]\n",
    "merge_x['model_overall'] = [merge_x.iloc[i]['Model_type_contig_amr'] if not pd.isna(merge_x.iloc[i]['Model_type_contig_amr']) else merge_x.iloc[i]['Reference Model Type_kma_amr'] for i in range(len(merge_x.index))]\n",
    "merge_x['Resistance_Mechanism_overall'] = [merge_x.iloc[i]['Resistance Mechanism_contig_amr'] if not pd.isna(merge_x.iloc[i]['Resistance Mechanism_contig_amr']) else merge_x.iloc[i]['Resistance Mechanism_kma_amr'] for i in range(len(merge_x.index))]\n",
    "\n",
    "merge_x.tail(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ORF_ID_contig_amr</th>\n",
       "      <th>Contig_contig_amr</th>\n",
       "      <th>Start_contig_amr</th>\n",
       "      <th>Stop_contig_amr</th>\n",
       "      <th>Orientation_contig_amr</th>\n",
       "      <th>Cut_Off_contig_amr</th>\n",
       "      <th>Pass_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_Bitscore_contig_amr</th>\n",
       "      <th>Best_Hit_ARO_contig_amr</th>\n",
       "      <th>Best_Identities_contig_amr</th>\n",
       "      <th>...</th>\n",
       "      <th>Promiscuous plasmid reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (chromosome or plasmid) reads_kma_sp</th>\n",
       "      <th>Unknown taxonomy (no genomic info) reads_kma_sp</th>\n",
       "      <th>ARO_kma</th>\n",
       "      <th>ARO_overall</th>\n",
       "      <th>Gene_Family_overall</th>\n",
       "      <th>Drug_Class_overall</th>\n",
       "      <th>model_overall</th>\n",
       "      <th>Resistance_Mechanism_overall</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9 # 11869 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_9</td>\n",
       "      <td>11869.0</td>\n",
       "      <td>12528.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Strict</td>\n",
       "      <td>410.0</td>\n",
       "      <td>439.884</td>\n",
       "      <td>arlr</td>\n",
       "      <td>99.54</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>207.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>arlr</td>\n",
       "      <td>arlr</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>fluoroquinolone antibiotic; acridine dye; disi...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8 # 10517 #...</td>\n",
       "      <td>NODE_47_length_13669_cov_51.565374_8</td>\n",
       "      <td>10517.0</td>\n",
       "      <td>11872.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>880.0</td>\n",
       "      <td>911.753</td>\n",
       "      <td>arls</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>304.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>arls</td>\n",
       "      <td>arls</td>\n",
       "      <td>major facilitator superfamily (MFS) antibiotic...</td>\n",
       "      <td>fluoroquinolone antibiotic; acridine dye; disi...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78 # 69097 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_78</td>\n",
       "      <td>69097.0</td>\n",
       "      <td>70452.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>850.0</td>\n",
       "      <td>885.174</td>\n",
       "      <td>mepa</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>426.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mepa</td>\n",
       "      <td>mepa</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>glycylcycline; tetracycline antibiotic</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79 # 70559 ...</td>\n",
       "      <td>NODE_4_length_124562_cov_63.896729_79</td>\n",
       "      <td>70559.0</td>\n",
       "      <td>70978.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>250.0</td>\n",
       "      <td>285.804</td>\n",
       "      <td>mepr</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mepr</td>\n",
       "      <td>mepr</td>\n",
       "      <td>multidrug and toxic compound extrusion (MATE) ...</td>\n",
       "      <td>glycylcycline; tetracycline antibiotic</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9 # 7790 # ...</td>\n",
       "      <td>NODE_41_length_18243_cov_57.681328_9</td>\n",
       "      <td>7790.0</td>\n",
       "      <td>8233.0</td>\n",
       "      <td>-</td>\n",
       "      <td>Perfect</td>\n",
       "      <td>270.0</td>\n",
       "      <td>301.212</td>\n",
       "      <td>mgra</td>\n",
       "      <td>100.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>mgra</td>\n",
       "      <td>mgra</td>\n",
       "      <td>ATP-binding cassette (ABC) antibiotic efflux p...</td>\n",
       "      <td>fluoroquinolone antibiotic; cephalosporin; pen...</td>\n",
       "      <td>protein homolog model</td>\n",
       "      <td>antibiotic efflux</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   ORF_ID_contig_amr  \\\n",
       "0  NODE_47_length_13669_cov_51.565374_9 # 11869 #...   \n",
       "1  NODE_47_length_13669_cov_51.565374_8 # 10517 #...   \n",
       "2  NODE_4_length_124562_cov_63.896729_78 # 69097 ...   \n",
       "3  NODE_4_length_124562_cov_63.896729_79 # 70559 ...   \n",
       "4  NODE_41_length_18243_cov_57.681328_9 # 7790 # ...   \n",
       "\n",
       "                       Contig_contig_amr  Start_contig_amr  Stop_contig_amr  \\\n",
       "0   NODE_47_length_13669_cov_51.565374_9           11869.0          12528.0   \n",
       "1   NODE_47_length_13669_cov_51.565374_8           10517.0          11872.0   \n",
       "2  NODE_4_length_124562_cov_63.896729_78           69097.0          70452.0   \n",
       "3  NODE_4_length_124562_cov_63.896729_79           70559.0          70978.0   \n",
       "4   NODE_41_length_18243_cov_57.681328_9            7790.0           8233.0   \n",
       "\n",
       "  Orientation_contig_amr Cut_Off_contig_amr  Pass_Bitscore_contig_amr  \\\n",
       "0                      -             Strict                     410.0   \n",
       "1                      -            Perfect                     880.0   \n",
       "2                      -            Perfect                     850.0   \n",
       "3                      -            Perfect                     250.0   \n",
       "4                      -            Perfect                     270.0   \n",
       "\n",
       "   Best_Hit_Bitscore_contig_amr Best_Hit_ARO_contig_amr  \\\n",
       "0                       439.884                    arlr   \n",
       "1                       911.753                    arls   \n",
       "2                       885.174                    mepa   \n",
       "3                       285.804                    mepr   \n",
       "4                       301.212                    mgra   \n",
       "\n",
       "   Best_Identities_contig_amr  ...  Promiscuous plasmid reads_kma_sp  \\\n",
       "0                       99.54  ...                               0.0   \n",
       "1                      100.00  ...                               0.0   \n",
       "2                      100.00  ...                               0.0   \n",
       "3                      100.00  ...                               0.0   \n",
       "4                      100.00  ...                               0.0   \n",
       "\n",
       "  Unknown taxonomy (chromosome) reads_kma_sp  \\\n",
       "0                                      207.0   \n",
       "1                                      304.0   \n",
       "2                                        0.0   \n",
       "3                                        0.0   \n",
       "4                                      106.0   \n",
       "\n",
       "  Unknown taxonomy (chromosome or plasmid) reads_kma_sp  \\\n",
       "0                                                0.0      \n",
       "1                                                0.0      \n",
       "2                                              426.0      \n",
       "3                                               36.0      \n",
       "4                                                0.0      \n",
       "\n",
       "   Unknown taxonomy (no genomic info) reads_kma_sp ARO_kma ARO_overall  \\\n",
       "0                                              0.0    arlr        arlr   \n",
       "1                                              0.0    arls        arls   \n",
       "2                                              0.0    mepa        mepa   \n",
       "3                                              0.0    mepr        mepr   \n",
       "4                                              0.0    mgra        mgra   \n",
       "\n",
       "                                 Gene_Family_overall  \\\n",
       "0  major facilitator superfamily (MFS) antibiotic...   \n",
       "1  major facilitator superfamily (MFS) antibiotic...   \n",
       "2  multidrug and toxic compound extrusion (MATE) ...   \n",
       "3  multidrug and toxic compound extrusion (MATE) ...   \n",
       "4  ATP-binding cassette (ABC) antibiotic efflux p...   \n",
       "\n",
       "                                  Drug_Class_overall          model_overall  \\\n",
       "0  fluoroquinolone antibiotic; acridine dye; disi...  protein homolog model   \n",
       "1  fluoroquinolone antibiotic; acridine dye; disi...  protein homolog model   \n",
       "2             glycylcycline; tetracycline antibiotic  protein homolog model   \n",
       "3             glycylcycline; tetracycline antibiotic  protein homolog model   \n",
       "4  fluoroquinolone antibiotic; cephalosporin; pen...  protein homolog model   \n",
       "\n",
       "  Resistance_Mechanism_overall  \n",
       "0            antibiotic efflux  \n",
       "1            antibiotic efflux  \n",
       "2            antibiotic efflux  \n",
       "3            antibiotic efflux  \n",
       "4            antibiotic efflux  \n",
       "\n",
       "[5 rows x 78 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_merged = merge_x\n",
    "final_merged.to_csv(f\"{output_dir}{prefix}.final_summary.tsv\", sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
